lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/TransactionReceipt.kt

```diff
@@ -70,29 +71,39 @@ sealed class TransactionReceipt {
     abstract val events: List<Event>
     abstract val messagesSent: List<MessageL2ToL1>
     abstract val executionResources: ExecutionResources
+    abstract val blockHash: Felt?
+    abstract val blockNumber: Int?
 
     val isAccepted: Boolean
         get() = (
             executionStatus == TransactionExecutionStatus.SUCCEEDED &&
                 (finalityStatus == TransactionFinalityStatus.ACCEPTED_ON_L1 || finalityStatus == TransactionFinalityStatus.ACCEPTED_ON_L2)
             )
-    abstract val isPending: Boolean
-}
 
-@Serializable
-sealed class ProcessedTransactionReceipt : TransactionReceipt() {
-    abstract val blockHash: Felt
-    abstract val blockNumber: Int
-    override val isPending: Boolean = false
-}
-
-@Serializable
-sealed class PendingTransactionReceipt : TransactionReceipt() {
-    override val isPending: Boolean = true
+    /**
+     * Checks if the transaction is in a pending block.
+     *
+     * This method provides accurate results for receipts obtained through [Provider.getTransactionReceipt].
+     * For receipts acquired using [Provider.getBlockWithReceipts], refer to [BlockWithReceipts.isPending].
+     *
+     * @return `true` if the transaction is pending; `false` otherwise.
+     */
+    val isPending: Boolean
+        get() = !hasBlockInfo && finalityStatus == TransactionFinalityStatus.ACCEPTED_ON_L2
```

Comment:
what if we get accepted_on_l1 ?

Reply:
It will not be a considered a pending receipt, as per the spec pending receipt should always have `ACCEPTED_ON_L2` finality status